lua/hbac/command/actions.lua

```diff
@@ -19,20 +19,27 @@ M.toggle_pin = function(bufnr)
 	return bufnr, pinned_state
 end
 
-M.set_all = function(pin_value)
+local set_all = function(pin_value)
 	local utils = require("hbac.utils")
 	local buflist = utils.get_listed_buffers()
+	local pins = utils.int_explode(vim.g.hbac_buffers or "")
+	if not pin_value then
+		vim.g.hbac_buffers = ""
```

Comment:
I think this line should have the early return. I think line 34 is overwriting this with the buflist equivalent even on `unpin_all`

Reply:
Yeah I there should have definitely been a return there (there is now). Tbh I forgot to test `unpin_all` (I did test `pin_all` ^^).